cli/render.go

```diff
@@ -75,21 +68,36 @@ func (r Render) toFile(c *RenderCommand, ec *errors.UIErrorContext) error {
 	return nil
 }
 
-func confirmOverwrite(c *RenderCommand) (bool, error) {
+func confirmOverwrite(c *RenderCommand, path string) (bool, error) {
+	// For non-interactive UIs, the value must be passed by flag.
+	if !c.ui.Interactive() {
+		return c.autoApproved, nil
+	}
+
+	if c.autoApproved || c.overwriteAll {
+		return true, nil
+	}
+
+	// For interactive UIs, we can do a y/n/a
 	for {
 		overwrite, err := c.ui.Input(&terminal.Input{
-			Prompt: "Output file exists, overwrite? [y/n] ",
+			Prompt: fmt.Sprintf("Output file %q exists, overwrite? [y/n/a] ", path),
 			Style:  terminal.WarningBoldStyle,
 		})
 		if err != nil {
 			return false, err
 		}
 		overwrite = strings.ToLower(overwrite)
-		if overwrite == "y" || overwrite == "n" {
-			return overwrite == "y", nil
+		switch overwrite {
+		case "a":
+			c.overwriteAll = true
+			fallthrough
```

Comment:
Is the idea to `return true, nil` like in the `y` case? If so I think it would be better to use the explicit `return` to avoid a potential bug if a new `case` gets added or the order changes.

Reply:
Side note not related to this PR: is the `for` loop used to guard against invalid options and ask the question again? If that's the case it would be nice to have `default` with some kind of `invalid option, please try again`.